src/main/java/Duke.java

```diff
@@ -1,10 +1,68 @@
 public class Duke {
+    protected static String REPLY_BAR = "%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%-%";
+    // Indents and wraps all of Duke's replies in the bars.
+    // Has a single line version...
+    private static void reply(String response) {
+        System.out.println("\n" + REPLY_BAR + "\n >>> " + response + "\n" + REPLY_BAR + "\n");
+    }
+    // and a multiline version as well.
+    private static void reply(String[] responses) {
+        System.out.println("\n" + REPLY_BAR);
+        for (String eachResponse : responses) {
+            System.out.println(" >>> " + eachResponse);
+        }
+        System.out.println(REPLY_BAR+"\n");
+    }
+
+    // Gracefully shuts down Duke.
+    // Doesn't do much now, but I'm guessing there will be file IO next time?
+    private static void stop() {
+        reply("Bye. Hope to see you again soon!");
+    }
+
+    // Main function. Main event loop happens here.
+    // The Parser instance is used to get user input and it returns a Command object.
+    // The Command object, which stores information from the user in an accessible interface,
+    // is then used to pass information to the switch statement that holds the command logic.
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String[] greetings = {"Hello! I'm Duke", "What can I do for you?"};
+        reply(greetings);
+
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser();
+        boolean endLoop = false;
+        Command latestCommand;
+
+        // MAIN EVENT LOOP:
+        while (!endLoop) {
+            parser.getUserInput();
+            latestCommand = parser.parseCommand();
+            // DEBUG:
+            // latestCommand.debug();
+
+            // SWITCH: Handles the latest command.
+            // The functions in other classes are written to return neat strings so that they can be wrapped by the
+            // 'reply' function. This is to shove all formatting logic into the functions to keep things neat here.
+            switch(latestCommand.getCommand()) {
+            case "bye": // Exit condition is here
+                stop();
+                endLoop = true;
+                break;
+            case "list": // List all tasks
+                reply(taskList.listTasks());
```

Comment:
I would like to suggest changing listTasks() to showTaskList()! I believe this alternative is clearer on task function since list can refer to both list (noun) and list (verb).